### DISCUSSION TITLE
some questions

### DISCUSSION THREAD

**Discussion by Sensei-CHO** [turtle0x1/LxdMosaic#477] [Q&A]
Hi! I've been using LxdMosaic for 3 days now and I love it!
It works correctly except for the Console(installed from clone) and it looks great!
But I wondered if you are going to implement using remote mysql server(for HA for example) and if you are going to implement more settings for the users(like quotas or limitation of instances, ram, cpu, etc...)

Sorry for my bad english and I hope to see more implementation in the project.

:wave:

**Comment by turtle0x1** [comment]
Hey glad your enjoying it, I saw your fork and thought it was strange you unwound lots of changes I guess the not working console explains it :D

There's been some changes round the console code so ill dig into it before I tag the next version!

You can use a remote mysql server by changing the values in `/var/www/LxdMosaic/.env` :)

Its on the long term agenda to add more cloud like control but RBAC (#327) comes first I think.

LXD supports a bunch of `limits` for `projects`, these limits can be set when you create a project on the CLI or LXDMosaic (see screenshot below) - do your needs differ?

![limits](https://user-images.githubusercontent.com/12494629/158015596-3154514c-4bf7-4ab3-96b9-c306060d7840.png)

**Comment by Sensei-CHO** [comment]
Thanks for your reply! I am not an expert with mysql but do you think I can share a remote mysql server for 2 LxdMosaic?(I want my LxdMosaic panel Highly Available) and I can't find what you showed in the screenshot where can I find it?

:wave:

**Reply by turtle0x1** [reply]
2 LXDMosaic instances is untested at this time, depending on how you want it configured will depend on the work that needs to go in.

Assuming you'll be doing round robin load balancing (I.E both are active at the same time), off the top of my head you'll need to deal with;

 - Pointing both instances at a central mysql server somewhere
 - Handling PHP `sessions` will require a central redis server somewhere
 - `/var/www/LxdMosaic/src/sensitiveData/certs` will need to be shared between all instances of LxdMosaic
 - You'll have to think about the node component, running two doesn't make a great deal of sense and will result in more sockets open than needed 

If this is for work i'm available for paid consulting / programming, you can get my email from `git log` but less than if the budget is less than £10K I wont be able to help unless its for a charity.
